projects/ngneat/helipopper/src/lib/helipopper.service.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  ApplicationRef,
+  ComponentFactoryResolver,
+  ElementRef,
+  Inject,
+  Injectable,
+  Injector,
+  NgZone,
+  TemplateRef,
+  Type
+} from '@angular/core';
+import { HELIPOPPER_CONFIG, HelipopperConfig } from './helipopper.types';
+import { HelipopperDirective } from './helipopper.directive';
+import { HelipopperOptions } from './helipopper-options';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HelipopperService {
+  constructor(
+    private appRef: ApplicationRef,
+    private zone: NgZone,
+    private resolver: ComponentFactoryResolver,
+    private hostInjector: Injector,
+    @Inject(HELIPOPPER_CONFIG) private config: HelipopperConfig
+  ) {}
+
+  open(host: ElementRef, helipopper: string | TemplateRef<any> | Type<any>, options?: Partial<HelipopperOptions>) {
+    let directive: HelipopperDirective = new HelipopperDirective(
+      host,
+      this.appRef,
+      this.zone,
+      this.resolver,
+      this.hostInjector,
+      this.config
+    );
+
+    directive.helipopper = helipopper;
+
+    directive.helipopperOptions = options?.helipopperOptions || {};
```

Comment:
Can we please make one single source of truth for the default options we use both here and in the directive? Such as placement, etc.

Reply:
done, i updated the new interface too by removing 'helipopper' in the properties wording as it's redundant with the interface name.